http3/server.go

```diff
@@ -205,6 +261,85 @@ func (s *Server) handleConn(sess quic.EarlySession) {
 	}
 }
 
+func (s *Server) handleControlStreams(sess quic.Session) (*ctrlStream, requestError) {
+	// send a SETTINGS frame
+	controlStreamOut, err := sess.OpenUniStream()
+	if err != nil {
+		return nil, newConnError(errorMissingSettings, err)
+	}
+	buf := bytes.NewBuffer([]byte{0})
+	(&settingsFrame{}).Write(buf)
+	controlStreamOut.Write(buf.Bytes())
+
+	controlStreamIn, err := sess.AcceptUniStream(context.Background())
+	if err != nil {
+		s.logger.Debugf("Accepting the incoming control stream failed.")
+		return nil, newConnError(errorMissingSettings, err)
+	}
+
+	pair := &ctrlStream{in: controlStreamIn, out: controlStreamOut}
+
+	br, ok := controlStreamIn.(byteReader)
+	if !ok {
+		br = &byteReaderImpl{controlStreamIn}
+	}
+	t, err := utils.ReadVarInt(br)
+	if t != 0x0 {
+		s.logger.Debugf("First stream must be a control stream")
+		return nil, newConnError(errorMissingSettings, err)
+	}
+
+	frame, err := parseNextFrame(controlStreamIn)
+	if err != nil {
+		s.logger.Debugf("Error encountered while parsing incoming frame")
+		return nil, newConnError(errorMissingSettings, err)
+	}
+	sf, ok := frame.(*settingsFrame)
+	if !ok {
+		s.logger.Debugf("First incoming frame parsed was not a settings frame")
+		return nil, newConnError(errorMissingSettings, nil)
+	}
+	// TODO: do something with the settings frame
+	s.logger.Debugf("Got settings frame: %+v", sf)
+
+	s.clients.Add(1)
+	connDone := make(chan struct{})
+	go func() {
+		defer ginkgo.GinkgoRecover()
```

Comment:
No.

Reply:
To which part? the `ginkgo.GinkgoRecover`, the go routine, clients waitgroup?

Reply:
The `GinkgoRecover`. Ginkgo is a testing framework and doesn't belong into non-test code.

Reply:
ok. there are other instance of ginkgorecover in that file, should I leave them there then? (they were there before)

Reply:
I'll take care of those. Must have been commit erroneously.

Reply:
See https://github.com/lucas-clemente/quic-go/pull/2372. Now with a custom linter to make sure this doesn't slip through again.